frontend/src/views/modelers/ModelersView.tsx

```diff
@@ -0,0 +1,107 @@
+/*******************************************************************************
+ * Copyright (c) 2019, 2020 Obeo.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+import { useLazyQuery, useQuery } from '@apollo/client';
+import { useMachine } from '@xstate/react';
+import gql from 'graphql-tag';
+import React, { useEffect } from 'react';
+import { useParams } from 'react-router-dom';
+import { ModelersEmptyView } from './ModelersEmptyView';
+import { ModelersErrorView } from './ModelersErrorView';
+import { ModelersLoadedView } from './ModelersLoadedView';
+import { ModelersLoadingView } from './ModelersLoadingView';
+import {
+  ErrorFetchingEvent,
+  FetchedModelersEvent,
+  ModelersUpdatedEvent,
+  ModelersViewContext,
+  ModelersViewEvent,
+  modelersViewMachine,
+} from './ModelersViewMachine';
+
+const getModelersQuery = gql`
+  query getModelers($projectId: ID!) {
+    viewer {
+      project(projectId: $projectId) {
+        modelers {
+          id
+          name
+          status
+        }
+      }
+    }
+  }
+`;
+
+export const ModelersView = () => {
+  const { projectId } = useParams();
+  const [{ value, context }, dispatch] = useMachine<ModelersViewContext, ModelersViewEvent>(modelersViewMachine);
+  const { modelers, message } = context;
+
+  // Load current project list
+  const { loading: modelersLoading, data: modelersData, error: modelersError } = useQuery(getModelersQuery, {
+    variables: { projectId },
+  });
+  useEffect(() => {
+    if (!modelersLoading) {
+      if (modelersError) {
+        const errorFetching = { type: 'ERROR_FETCHING', message: modelersError.message } as ErrorFetchingEvent;
+        dispatch(errorFetching);
+      } else if (modelersData) {
+        let { modelers } = modelersData?.viewer.project;
+        const fetchedProjects = { type: 'FETCHED_MODELERS', modelers } as FetchedModelersEvent;
+        dispatch(fetchedProjects);
+      }
+    }
+  }, [modelersLoading, modelersData, modelersError, dispatch]);
+
+  // Setup callback to update project list when invoked
+  const [getModelers, { loading, error, data }] = useLazyQuery(getModelersQuery);
+  useEffect(() => {
+    if (!loading) {
+      if (error) {
+        const errorFetching = { type: 'ERROR_FETCHING', message: error.message } as ErrorFetchingEvent;
+        dispatch(errorFetching);
+      } else if (data) {
```

Comment:
data cannot be null or undefined here, no need for `?`. No need for the else here too, we can have both.